unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2031,16 +2036,25 @@ abstract public class BaseFPSAgentController {
             if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(objectId)) {
                 throw new ArgumentException($"objectId: {objectId} is not the objectId on any object in the scene!");
             }
-
-            // if object is in the scene and visible, assign it to 'target'
-            SimObjPhysics target = getInteractableSimObjectFromId(objectId: objectId, forceVisible: forceAction);
+            
+            SimObjPhysics sop = getSimObjectFromId(objectId);
+            if (sop == null) {
+                throw new NullReferenceException("Object with id'" + objectId + "' is null");
+            }
+            
+            SimObjPhysics[] interactable;
+            bool visible = GetAllVisibleSimObjPhysics(camera: this.m_Camera, maxDistance: this.maxVisibleDistance, out interactable, filterSimObjs: new List<SimObjPhysics> { sop }).Length == 1;
```

Comment:
It seems like we should have a nicer way of doing this, is the reason we use `GetAllVisibleSimObjPhysics` rather than `isSimObjVisible` because we need to account for the difference between `GetAllVisibleSimObjPhysicsCollider` and `GetAllVisibleSimObjPhysicsDistance`?

Reply:
Yes, currently the default visibility schema uses the capsule collider based approach. Eric and I were talking about potentially switching over the default to use purely a distance based check like in `isSimObjVisible` as the new default, but that will be addressed in a different PR down the line.